src/test/ui/never_type/fallback-closure-wrap.rs

```diff
@@ -0,0 +1,17 @@
+// check-pass
```

Comment:
Can you add a comment for this test?

And add the example in the OP (for projection outputs) as a test?